File: BOJ/15650.swift
import Foundation

let input = readLine()!.split(separator: " ").map{Int(String($0))!}

var v = [Int]()
func dfs(_ idx: Int ) {
    if v.count == input[1] {
        print(v.map{String($0)}.joined(separator: " "))
    }
    for i in idx..<input[0]+1 {
        v.append(i)
        dfs(i+1)
        v.removeLast()
    }
}
dfs(1)

File: BOJ/15650.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
v = []

def dfs(v, idx):
    if len(v) == m:
        for i in range(m):
            print(v[i], end=" ")
        print()
        return
    for i in range(idx,n+1):
        v.append(i)
        dfs(v,i+1)
        v.pop()

dfs(v,1)
